alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultConfigurationAccessor.java

```diff
@@ -331,9 +331,11 @@ public ConfigurationModel updateConfiguration(Long descriptorConfigId, Collectio
         if (configuredFields != null && !configuredFields.isEmpty()) {
             List<FieldValueEntity> fieldValuesToSave = new ArrayList<>(configuredFields.size());
             for (ConfigurationFieldModel configFieldModel : configuredFields) {
-                Long fieldId = getFieldIdOrThrowException(configFieldModel.getFieldKey());
+                String fieldKey = configFieldModel.getFieldKey();
+                Long fieldId = getFieldIdOrThrowException(fieldKey);
+                boolean isSensitive = isFieldSensitive(fieldKey);
```

Comment:
We weren't checking if the database defined the field as sensitive we were trusting what was being passed into update.  In the OAuth case the credentials store would create the ConfigurationFieldModels with the create method and the sensitive field would not be set for fields that were sensitive in the database.  Here we look up what the database has defined in order to do the right thing.